### PR TITLE
db decoder functions

### PR DESCRIPTION
**Pull request by leoscholl**
add db functions for fetching and loading decoders

### CODE REVIEW

File: aopy/data/db.py [hunk 1]
```diff
@@ -114,6 +115,34 @@ def lookup_bmi_sessions(bmi_task_name='bmi control', **kwargs):
...
+
+    Returns:
+        list: list of Decoder records matching the query
+    '''
+    if id is not None and isinstance(id, list):
+        kwargs['id__in'] = id
+    elif id is not None:
+        kwargs['id'] = id
+    if parent_id is not None:
+        kwargs['entry_id'] = parent_id
```
Comment (rcanfield2) [inline aopy/data/db.py:141]: Do you need to do the same condition if parent_id is or is not a list?

File: aopy/data/db.py [hunk 2]
```diff
@@ -414,6 +443,96 @@ def get_db_object(self):
         '''
         return self.record
 
+class BMI3DDecoder():
+    '''
+    Wrapper for BMI3D Decoder objects
+    '''
+
```
Comment (rcanfield2) [inline aopy/data/db.py:450]: Can you add a little more information about what this class does with BMI3D decoder objects?

File: aopy/data/db.py [hunk 3]
```diff
@@ -414,6 +443,96 @@ def get_db_object(self):
...
+        Constructor
+
+        Args:
+            task_entry (TaskEntry): a database task entry from BMI3D
+            dbname (str, optional): name of the database to connect to. Defaults to 'default'.
+        '''
+        self.dbname = dbname
+        self.record = decoder
+        self.id = self.record.id
+        self.parent = lookup_sessions(id=decoder.entry_id)[0]
```
Comment (rcanfield2) [inline aopy/data/db.py:462]: Can decoder.entry_id be empty or None?

File: aopy/data/db.py [hunk 4]
```diff
@@ -414,6 +443,96 @@ def get_db_object(self):
...
+        self.name = self.record.name
+        
+    def __repr__(self):
+        '''
+        Call __repr__ on the database record
+
+        Returns:
+            str: repr
+        '''
+        return f"Decoder[{self.id}:{self.name}] trained from {repr(self.parent)}"
```
Comment (rcanfield2) [inline aopy/data/db.py:474]: Same as above comment if applicable

**Review by rcanfield2 — APPROVED** [review]

File: aopy/data/db.py (hunk 1 above)
Comment (leoscholl) [reply aopy/data/db.py:141]: i don't think it makes as much sense to lookup decoders from multiple parents..?

File: aopy/data/db.py (hunk 2 above)
Comment (leoscholl) [reply aopy/data/db.py:450]: it's a wrapper :)

File: aopy/data/db.py (hunk 3 above)
Comment (leoscholl) [reply aopy/data/db.py:462]: good question, i don't think so but it can't hurt to check

**Pull request merged**